### PR TITLE
Revert "6318: fixed mention complete with mouse clicking"

### LINKED ISSUE #6318: Can not complete a mention with the mouse
1538-446-max

i can't click on the name that i want in nextcloud talk. on chrome.

### PR DESCRIPTION
**Pull request by marcoambrosini**
Reverts nextcloud/spreed#6789

Fix #6318

### CODE REVIEW

File: src/components/NewMessageForm/AdvancedInput/AdvancedInput.vue [hunk 1]
```diff
@@ -248,26 +251,26 @@ export default {
...
+				}
+			}, 200)
 		},
-		closePanel() {
-			if (this.$refs.at) {
-				this.$refs.at.closePanel()
-			}
+
+		onFocus() {
+			clearTimeout(this.blurTimer)
```
Comment (nickvergessen) [inline src/components/NewMessageForm/AdvancedInput/AdvancedInput.vue:271]: check for null?

Comment (marcoambrosini) [reply src/components/NewMessageForm/AdvancedInput/AdvancedInput.vue:271]: you can clear a null timeout. Shouldn't throw anything

**Review by danxuliu — COMMENTED** [review]
Tested and works :+1:

However, @marcoambrosini when reverting please describe why you are reverting :-) In this case I guess it is due to the event listener being set again and again when the input is focused, as it uses an arrow function, but better to have it explicit in the pull request description.

Could you also explain why the original code (before https://github.com/nextcloud/spreed/pull/6789) failed and why your change fixes it? I have tested just a few times and it worked, but it would be good to know where was the problem to test it in more detail, as the original code also worked some times.

Similarly, @nikola-gladovic, could you explain why your code fixed the issue too so both approaches can be compared?

Thanks :)

**Comment by marcoambrosini** [comment]
@danxuliu the commit message explanation is not good enough?? xD

File: src/components/NewMessageForm/AdvancedInput/AdvancedInput.vue [hunk 2]
```diff
@@ -213,6 +214,8 @@ export default {
 		return {
 			text: '',
 			autoCompleteMentionCandidates: [],
+			blurTimer: null,
+
```
Comment (nickvergessen) [inline src/components/NewMessageForm/AdvancedInput/AdvancedInput.vue:218]: CS checker is dying a little inside

**Comment by danxuliu** [comment]
> @danxuliu the commit message explanation is not good enough?? xD

Ah, sorry, I should have looked there, indeed ;-)

Relying on timeouts to do something is flaky, though... It would be interesting to understand why Nikola's approach also fixed the issue, as it could be a safer approach once the duplicated listeners are fixed :thinking:

**Comment by nickvergessen** [comment]
> Relying on timeouts to do something is flaky, though... It would be interesting to understand why Nikola's approach also fixed the issue, as it could be a safer approach once the duplicated listener is fixed thinking

But that is what the lib author mentions as solution:
https://github.com/fritx/vue-at/issues/114#issuecomment-565777450

**Comment by marcoambrosini** [comment]
Done, also fine tuned messages spacing

https://user-images.githubusercontent.com/26852655/150321685-d6fe4f8e-575d-4526-a0bc-18a0ca1acb91.mov

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**